test/src/unit-cppapi-hilbert.cc

```diff
@@ -1554,8 +1554,8 @@ TEST_CASE(
 
   // Write first fragment
   std::vector<int32_t> buff_a = {2, 3, 1, 4};
-  std::vector<float> buff_d1 = {0.1f, 0.1f, 0.4f, 0.5f};
-  std::vector<float> buff_d2 = {0.3f, 0.1f, 0.2f, 0.4f};
+  std::vector<float> buff_d1 = {0.1f, 0.100599997f, 0.4f, 0.5f};
+  std::vector<float> buff_d2 = {0.3f, 0.109373093f, 0.2f, 0.4f};
   write_2d_array<float, float>(
```

Comment:
Comment for the reviewers, this is an obscure test IMO. We set the output buffers to 0, so that we are always forced to split if there is data inside of the tile we are looking at. We will then split until the range is unsplittable and get the data for that range, which in all likelihood there won't be any and process all unsplittable ranges until we get lucky, find one data point, see that it doesn't fit inside of our 0 sized buffers and return unsplittable. Since we have 64 bits for the Hilbert buckets, getting lucky might take a while... This test was configured to hit that 0.1, 0,1 value when it reached unsplittable but since we have corrected the splitting algorithm, this value is now the one in the test.